pipeline/api/cloud.py

```diff
@@ -421,6 +422,7 @@ def upload_pipeline(
         public: bool = False,
         description: str = "",
         tags: Set[str] = None,
+        environment: Optional[Union[PipelineCloudEnvironment, str]] = None,
```

Comment:
The name here is a bit confusing, perhaps `environment_object_or_id` or something to that effect might be better.

Reply:
I try to avoid putting the types into the name as that information is already encoded in the type and/or the docstring. But it's not a particularly strong preference so am happy to change if you'd like.

Reply:
another option that I've seen used is to create a type alias, eg something like:

```python
EnvironmentObjectOrId = Union[PipelineCloudEnvironment, str]

...
environment: Optional[EnvironmentObjectOrId] = None,
...
```

Reply:
Yeah, that would make the annotation more readable.

With Python 3.10 it could become even nicer: `PipelineCloudEnvironment | str | None`.

Reply:
I've factored out the type.